src/query/storages/system/src/columns_table.rs

```diff
@@ -168,3 +219,28 @@ impl ColumnsTable {
         Ok(rows)
     }
 }
+
+pub fn find_eq_table(expr: &Expr<String>, visitor: &mut impl FnMut(&str, &Scalar)) {
+    match expr {
+        Expr::Constant { .. } => {}
+        Expr::ColumnRef { .. } => {}
+        Expr::Cast { expr, .. } => {
+            find_eq_table(expr, visitor);
+        }
+        Expr::FunctionCall { function, args, .. } => {
+            if function.signature.name == "eq" {
```

Comment:
Only check equal function may produce some unintended results, for example
```sql
where table='tb1' or table !='tb2';
```

Reply:
Yes, this maybe cause some other bad case like :

```sql
where table='tb1' and table in ('tb2', 'tb3', 'tb4', 'tb5')
```

So I think filter only contains `eq` `and` `or` do this optimize.